packages/next-intl/test/middleware/middleware.test.tsx

```diff
@@ -1765,6 +1765,23 @@ describe('domain-based routing', () => {
       );
     });
 
+    it('prioritizes the default locale of a domain', () => {
+      const m = createIntlMiddleware({
+        defaultLocale: 'en',
+        locales: ['en', 'fr'],
+        domains: [
+          {
+            defaultLocale: 'fr',
+            domain: 'ca.example.com'
+          }
+        ]
+      });
+      m(createMockRequest('/', 'de', 'http://ca.example.com'));
+      expect(MockedNextResponse.redirect.mock.calls[0][0].toString()).toBe(
+        'http://ca.example.com/fr'
+      );
+    });
```

Comment:
Note that the `defaultLocale` of a domain is used if no other locale matches better. However, note that if a domain supports multiple locales, the best-matching one will be selected based on the `accept-language` header.